handlers/handlers.go

```diff
@@ -25,29 +39,157 @@ func setStatusCode(req *http.Request, w http.ResponseWriter, err error) {
+// getCookiePreferencePage talks to the renderer to get the cookie preference page
+func getCookiePreferencePage(ctx context.Context, w http.ResponseWriter, req *http.Request, rendC RenderClient, cp cookies.Policy) error {
+	var err error
+	m := mapper.CreateCookieSettingPage(cp)
+	b, err := json.Marshal(m)
+	if err != nil {
+		log.Event(ctx, "unable to marshal cookie preferences", log.Error(err))
+		setStatusCode(req, w, err)
+		return err
+	}
+
+	templateHTML, err := rendC.Do("cookies-preferences", b)
+	if err != nil {
+		log.Event(ctx, "getting template from renderer cookies-preferences failed", log.Error(err))
+		setStatusCode(req, w, err)
+		return err
+	}
+	if _, err := w.Write(templateHTML); err != nil {
+		log.Event(ctx, "error on write of cookie template", log.Error(err))
+		setStatusCode(req, w, err)
+	}
+	return err
+}
+
+// isProtectedCookie is a helper function that checks if a cookie is protected or not
+func isProtectedCookie(stringToFind string, slice [5]string) bool {
+	for _, element := range slice {
+		if element == stringToFind {
+			return true
+		}
+	}
+	return false
+}
+
+// removeNonProtectedCookies sets cookies which replace any existing ones with an instant expiry; which removes them.
+// This will not remove any protected cookies
+func removeNonProtectedCookies(w http.ResponseWriter, req *http.Request) {
+	for _, cookie := range req.Cookies() {
+		if !isProtectedCookie(cookie.Name, protectedCookies) {
+			cookie := &http.Cookie{
+				Name:    cookie.Name,
+				Value:   "",
+				Path:    "/",
+				Expires: time.Unix(0, 0),
+
+				HttpOnly: false,
+			}
+			http.SetCookie(w, cookie)
+		}
+	}
+}
+
+// AcceptAll handler for setting all cookies to enabled then refresh the page. when JS has been disabled
+// Example usage; JavaScript disabled.
+func AcceptAll() http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		acceptAll(w, req)
+	}
+}
+
+// Read Handler
+func Read(rendC RenderClient) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		read(w, req, rendC)
+	}
+}
+
+// Edit Handler
+func Edit(rendC RenderClient) http.HandlerFunc {
 	return func(w http.ResponseWriter, req *http.Request) {
-		helloWorld(w, req, cfg)
+		edit(w, req, rendC)
 	}
 }
 
-func helloWorld(w http.ResponseWriter, req *http.Request, cfg config.Config) {
+// acceptAll handler for accepting all possible cookies
+func acceptAll(w http.ResponseWriter, req *http.Request) {
 	ctx := req.Context()
-	greetingsModel := mapper.HelloModel{Greeting: "Hello", Who: "World"}
-	m := mapper.HelloWorld(ctx, greetingsModel, cfg)
+	cp := cookies.Policy{
+		Essential: true,
+		Usage:     true,
+	}
+	reqUrl, err := url.Parse(req.URL.Path)
+	if err != nil {
+		log.Event(ctx, "unable to parse url", log.Error(err))
+		setStatusCode(req, w, err)
+		return
+	}
+	cookies.SetPolicy(w, cp, reqUrl.Hostname())
+	cookies.SetPreferenceIsSet(w, reqUrl.Hostname())
+	referer := req.Header.Get("Referer")
```

Comment:
Very very picky comment but we have a `headers` package in the `dp-api-clients-go` maybe a but overkill but consider adding a new getter to that package for this header. It will help ensure consistency.

https://github.com/ONSdigital/dp-api-clients-go/blob/master/headers/headers.go